app/src/main/java/com/example/recipeapp/viewmodels/LoginViewModel.java

```diff
@@ -1,46 +1,33 @@
 package com.example.recipeapp.viewmodels;
 
-import android.content.Context;
-import android.content.Intent;
-import android.util.Log;
-import android.view.View;
-import android.widget.Toast;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
 
-import androidx.databinding.BaseObservable;
-
-import com.example.recipeapp.MainActivity;
-import com.example.recipeapp.SignUpActivity;
+import com.parse.LogInCallback;
+import com.parse.ParseException;
 import com.parse.ParseUser;
 
-public class LoginViewModel extends BaseObservable {
-
-    private static final String TAG = "Logging in: ";
+public class LoginViewModel extends ViewModel {
 
     private String username;
     private String password;
 
-    public LoginViewModel() {
 
+    public MutableLiveData<Boolean> startSignUp = new MutableLiveData<>();
+    public MutableLiveData<ParseException> isLogInSuccessful = new MutableLiveData<>();
+
+    public LoginViewModel() {
     }
 
-    public void login(View view) {
+    public void login() {
         ParseUser.logInInBackground(
                 username,
                 password,
-                (user, e) -> {
-                    Context context = view.getContext();
-                    if(e != null) {
-                        Log.i(TAG, "Issue with login" + e);
-                        Toast.makeText(context, "Unable to login", Toast.LENGTH_SHORT).show();
-                        return;
-                    }
```

Comment:
Lets log exception in backend and pass just error code/enum maybe to the frontend.